packages/smooth_app/lib/pages/scan/lifecycle_manager.dart

```diff
@@ -22,29 +25,16 @@ class LifeCycleManager extends StatefulWidget {
 
 class LifeCycleManagerState extends State<LifeCycleManager>
     with WidgetsBindingObserver {
-  double visibleFraction = 100.0;
   AppLifecycleState appLifecycleState = AppLifecycleState.resumed;
 
-  void checkLifeCycle() {
-    if (appLifecycleState == AppLifecycleState.inactive ||
-        visibleFraction == 0.0) {
-      widget.onPause.call();
-    } else if (appLifecycleState == AppLifecycleState.resumed &&
-        visibleFraction > 0.0) {
-      widget.onResume.call();
-    }
-  }
-
   @override
   void initState() {
     super.initState();
     WidgetsBinding.instance!.addObserver(this);
-  }
 
-  @override
-  void dispose() {
-    WidgetsBinding.instance!.removeObserver(this);
-    super.dispose();
+    if (widget.onStart != null) {
+      WidgetsBinding.instance!.addPostFrameCallback((_) => widget.onStart!());
```

Comment:
That means that just using `onResume` is not good enough, we need a boost for the very first time, right?

Reply:
Yes exactly.
And since this Widget is independent of the camera lifecycle, I prefer to differentiate this step

Reply:
Please add a comment then, like `// Boost needed for the very first time, as experiments showed`